### DISCUSSION TITLE
Should we be removing the client code from the repo?

### DISCUSSION THREAD

**Discussion by Jamstah** [distribution/distribution#3791] [Ideas]
From what I understand, the client in this repo is unsupported.

Should we remove it from main? If anyone is using it, they can use the 2.8 branch until they move off it, which is recommended because it is not maintained anyway.

**Comment by milosgajdos** [comment]
We would have to check some data (GH BigQuery?) to see if the client code packages are imported by some other tools just so we gauge if there is anyone whose code depends on this for whatever reason 😬

**Reply by Jamstah** [reply]
I may have the wrong end of the stick anyway - we use it ourselves for the proxy implementation, so we wouldn't be able to get rid of it unless we pulled in a different implementation for that, surely?

In terms of 3rd parties, it looks like argo uses it: https://github.com/argoproj-labs/argocd-image-updater/blob/master/pkg/registry/client.go

Everything else in my search (`github.com/distribution/distribution/v3/registry/client`) seemed to be vendoring.

**Reply by milosgajdos** [reply]
Well at least `moby` uses it too https://github.com/moby/moby/blob/3330fc2e9320e46d345132f07ec0ea7b42029e47/registry/auth.go#L9
